### PR TITLE
Add support of ESP RainMaker to Arduino.

### PR DESCRIPTION
**Pull request by sweetymhaiske**
1. Provides ESP RainMaker functionality to Arduino.
2. Added ESP RainMaker libs, built upon esp-rainmaker GitHub repo.
3. Arduino RainMaker library, allows Arduino user to play with ESP RainMaker.
4. This provides Arduino users to realize their IoT ideas faster using phone apps, without making any configuration to the cloud.

### CODE REVIEW

File: libraries/RainMaker/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+The key features of ESP RainMaker are:
+
+1. Ability to define own devices and parameters, of any type, in the firmware.
+2. Zero configuration required on the Cloud.
+3. Phone apps that dynamically render the UI as per the device information.
+
+This RainMaker library is built using esp-rainmaker component.
+
+#### The link for esp-rainmaker repository
```
Comment (shahpiyushv) [inline libraries/RainMaker/README.md:18]: Does the header need to be so long? Just Repository Source?

File: libraries/RainMaker/examples/RMakerDevice/RMakerDevice.ino [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    my_node = RMaker.initNode("ESP RainMaker Node");
+
+    //Standard switch device
+    my_switch.addCb(write_callback);
+    
+    //Creating custom fan device
+    my_device.addNameParam();
+    my_device.addPowerParam(DEFAULT_POWER_MODE);
+    my_device.addSpeedParam(DEFAULT_FAN_SPEED);
+    my_device.assignPrimaryParam(my_device.getParamByName("Power"));
```
Comment (me-no-dev) [inline libraries/RainMaker/examples/RMakerDevice/RMakerDevice.ino:73]: is `Power` a custom string or is it something coming from RainMaker itself? If so, it should be defined somewhere in RMaker

**Comment by me-no-dev** [comment]
looks generally OK. left one note. Why was it necessary to expose init/deinit for WiFi Generic?

File: libraries/RainMaker/examples/RMakerDevice/RMakerDevice.ino (hunk 2 above)
Comment (sweetymhaiske) [reply libraries/RainMaker/examples/RMakerDevice/RMakerDevice.ino:72]: For default cases, it is coming from RainMaker. Yes, it is fixed in a newer commit.

**Comment by me-no-dev** [comment]
closing as no longer needed :)

**Pull request closed without merging**